tap_google_sheets/streams.py

```diff
@@ -1,74 +1,506 @@
+import json
+import os
+import time
+import re
 from collections import OrderedDict
+import urllib.parse
+import singer
+from singer import metrics, metadata, Transformer, utils
+from singer.utils import strptime_to_utc, strftime
+from singer.messages import RecordMessage
+import tap_google_sheets.transform as internal_transform
+import tap_google_sheets.schema as schema
 
-# streams: API URL endpoints to be called
-# properties:
-#   <root node>: Plural stream name for the endpoint
-#   path: API endpoint relative path, when added to the base URL, creates the full path,
-#       default = stream_name
-#   key_properties: Primary key fields for identifying an endpoint record.
-#   replication_method: INCREMENTAL or FULL_TABLE
-#   replication_keys: bookmark_field(s), typically a date-time, used for filtering the results
-#       and setting the state
-#   params: Query, sort, and other endpoint specific parameters; default = {}
-#   data_key: JSON element containing the results list for the endpoint;
-#       default = root (no data_key)
-
-# file_metadata: Queries Google Drive API to get file information and see if file has been modified
-#    Provides audit info about who and when last changed the file.
-FILE_METADATA = {
-    "api": "files",
-    "path": "files/{spreadsheet_id}",
-    "key_properties": ["id"],
-    "replication_method": "INCREMENTAL",
-    "replication_keys": ["modifiedTime"],
-    "params": {
+LOGGER = singer.get_logger()
+
+# Currently syncing sets the stream currently being delivered in the state.
+# If the integration is interrupted, this state property is used to identify
+#  the starting point to continue from.
+# Reference: https://github.com/singer-io/singer-python/blob/master/singer/bookmarks.py#L41-L46
+def update_currently_syncing(state, stream_name):
+    """
+    Currently syncing sets the stream currently being delivered in the state.
+    If the integration is interrupted, this state property is used to identify
+     the starting point to continue from.
+    Reference: https://github.com/singer-io/singer-python/blob/master/singer/bookmarks.py#L41-L46
+    """
+    if (stream_name is None) and ("currently_syncing" in state):
+        del state["currently_syncing"]
+    else:
+        singer.set_currently_syncing(state, stream_name)
+    singer.write_state(state)
+
+def write_schema(catalog, stream_name):
+    stream = catalog.get_stream(stream_name)
+    schema = stream.schema.to_dict()
+    try:
+        singer.write_schema(stream_name, schema, stream.key_properties)
+        LOGGER.info('Writing schema for: {}'.format(stream_name))
+    except OSError as err:
+        LOGGER.info('OS Error writing schema for: {}'.format(stream_name))
+        raise err
+
+def write_record(stream_name, record, time_extracted, version=None):
```

Comment:
Add function comment, why do we need to check the version?

Reply:
Added comment.